packages/engine/src/avatar/functions/updateVRMRetargeting.ts

```diff
@@ -55,23 +49,10 @@ export const updateVRMRetargeting = (vrm: VRM, deltaTime: number) => {
 
       // Move the mass center of the VRM
       if (boneName === 'hips') {
-        /** @todo for some reason this breaks loop animations */
         const boneWorldPosition = rigBoneNode.getWorldPosition(_boneWorldPos)
-
-        const boneEntity = boneNode.entity
-        if (!boneEntity) continue
-
-        const parentEntity = getComponent(boneEntity, EntityTreeComponent)?.parentEntity
-        if (!parentEntity) continue
-
-        const parentBoneNode = getComponent(parentEntity, BoneComponent)
-        if (!parentBoneNode) continue
-
-        computeTransformMatrix(parentEntity)
-        parentBoneNode.updateMatrixWorld(true)
-
-        const parentWorldMatrix = parentBoneNode.matrixWorld
-        const localPosition = boneWorldPosition.applyMatrix4(parentWorldMatrix.clone().invert())
+        boneNode.parent!.updateWorldMatrix(true, false)
```

Comment:
why are we reverting from ECS language to threejs?

The two clone() calls are also going to be unnecessary and expensive

Reply:
Removed one of the clone calls, but I can't find a way around cloning the matrix world right now.

Reply:
module scope
`const mat4 = new Matrix4()`

function scope
`const localPosition = boneWorldPosition.applyMatrix4(mat4.copy(parentWorldMatrix).invert())`

Reply:
Thanks. Don't know why I didn't realize that 🤦